lib/User_Proxy.php

```diff
@@ -263,13 +283,35 @@ public function hasUserListings() {
 	 */
 	public function countUsers() {
 		$users = false;
-		foreach($this->backends as $backend) {
-			$backendUsers = $backend->countUsers();
-			if ($backendUsers !== false) {
-				$users += $backendUsers;
+		foreach($this->backends as $configPrefix => $backend) {
+			try {
+				$backendUsers = $backend->countUsers();
+				if ($backendUsers !== false) {
+					$users += $backendUsers;
+				}
+			} catch (\OC\ServerNotAvailableException $ex) {
+				$this->handleServerNotAvailable($configPrefix, $ex);
+				continue;
 			}
 		}
 		return $users;
 	}
 
+	/**
+	 * Show a log message only once per configuration prefix in order to prevent log flooding
+	 * @param string $configPrefix the configuration prefix for the LDAP connection
+	 * @param \Exception $ex the exception thrown
+	 */
+	protected function handleServerNotAvailable($configPrefix, $ex) {
+		static $messages = array();
```

Comment:
It isn't needed to be a CappedMemoryCache. I expect a maximum of 10, or even 20 different servers. It won't consume a lot of memory.

> Would be better to define this as static class attribute instead of hidden inside the function.

It doesn't add any value to the class, just noise, that's why I'd rather keep it inside the function scope. It's for "personal use" so other functions aren't expected to use that variable.

I think someone will need to break the tie 😄